src/pretix/presale/checkoutflow.py

```diff
@@ -301,15 +318,17 @@ def post(self, request):
                 self.cart_session['customer'] = request.customer.pk
                 self.cart_session['customer_cart_tied_to_login'] = True
                 return redirect(self.get_next_url(request))
-            elif self.login_form.is_valid():
+            elif "login-sso-data" in self.request.POST and self._handle_sso_login():
```

Comment:
Should the `_handle_sso_login` be done in the if statement or would it be better to explicitly just check for the `"login-sso-data" in self.request.POST` and then check `self._handle_sso_login()` separately and output some error? Otherwise it will just skip to the next elif/else, which is probably not what we want? So something like:

```suggestion
            elif "login-sso-data" in self.request.POST:
                if not self._handle_sso_login():
                    messages.error(self.request, _("Login failed."))
```